packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -1020,9 +1017,7 @@ class FlutterPlugin implements Plugin<Project> {
         if (localProperties == null) {
             localProperties = readPropertiesIfExist(new File(project.projectDir.parentFile, "local.properties"))
         }
-        String result = project.hasProperty(name) ? project.property(name) : null
-        result = result ?: localProperties?.getProperty(name)
-        return result ?: defaultValue
+        return project.findProperty(name) ?: localProperties?.getProperty(name, defaultValue)
```

Comment:
`findProperty`:Returns the value of the given property or null if not found. This method locates a property as follows:
1. If this project object has a property with the given name, return the value of the property.
2. If this project has an extension with the given name, return the extension.
3. If this project's convention object has a property with the given name, return the value of the property.
4. If this project has an extra property with the given name, return the value of the property.
5. If this project has a task with the given name, return the task.
Search up through this project's ancestor projects for a convention property or extra property with the given name.
6. If not found, null value is returned.

Reply:
potentially breaking because of the exact value this solution brings. Existing conflicting named properties that are in the newly discovered locations could break builds. I am not 100% opposed but it worth looking at.